src/pilot/assign_role.py

```diff
@@ -296,48 +296,116 @@ def get_raid_controller_ids(drac_client):
     return raid_controller_ids
 
 
-def define_controller_logical_disks(drac_client, raid_controller_ids):
-    if len(raid_controller_ids) > 1:
-        LOG.critical(
-            "Found more than one RAID controller:\n  {}".format(
-                "\n  ".join(raid_controller_ids)))
-        return None
+def define_controller_logical_disks(drac_client, raid_controllers):
+    all_physical_disks = drac_client.list_physical_disks()
 
-    raid_10_logical_disk = define_single_raid_10_logical_disk(
-        drac_client, raid_controller_ids[0])
+    # Get the drives controlled by the RAID controller
+    raid_cntlr_physical_disks = {}
+    for disk in all_physical_disks:
+        if disk.controller in raid_controllers:
+            if disk.controller in raid_cntlr_physical_disks:
```

Comment:
this can be shortened by using defaultdict(list) for raid_cntlr_physical_disks

Reply:
@mpardhi23, you may find https://stackoverflow.com/questions/26367812/appending-to-list-in-python-dictionary helpful.

Reply:
Done